src/shogun/distributions/classical/GaussianDistributionNew.h

```diff
@@ -0,0 +1,175 @@
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * The views and conclusions contained in the software and documentation are those
+ * of the authors and should not be interpreted as representing official policies,
+ * either expressed or implied, of the Shogun Development Team.
+ *
+ */
+
+#ifndef GAUSSIANDISTRIBUTIONNEW_H
+#define GAUSSIANDISTRIBUTIONNEW_H
+
+#include <shogun/lib/config.h>
+
+#ifdef HAVE_EIGEN3
+
+#include <shogun/distributions/classical/ProbabilityDistribution.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/mathematics/Math.h>
+
+namespace shogun
+{
+/** TODO commenting properly the new fields
+ */
+
+/** @brief Dense version of the well-known Gaussian probability distribution,
+ * defined as
+ * \f[
+ * \mathcal{N}_x(\mu,\Sigma)=
+ * \frac{1}{\sqrt{|2\pi\Sigma|}}
+ * \exp\left(-\frac{1}{2}(x-\mu)^T\Sigma^{-1}(x-\mu)\right)
+ * \f]
+ *
+ * The implementation represents the covariance matrix \f$\Sigma \f$, as
+ * Cholesky factorisation, such that the covariance can be computed as
+ * \f$\Sigma=LL^T\f$.
+ */
+
+class CGaussianDistributionNew: public CProbabilityDistribution
+{
+public:
+	/** Default constructor */
+	CGaussianDistributionNew();
+
+	/** Constructor for which takes Gaussian mean and its covariance matrix.
+	 * It is also possible to pass a precomputed matrix factor of the specified
+	 * form. In this case, the factorization is not explicitly computed.
+	 *
+	 * @param mean mean of the Gaussian
+	 * @param cov covariance of the Gaussian, or covariance factor
+	 * @param cov_is_cholesky whether cov is a Cholesky factorization of the covariance or not
+	 * (default is false). If false, the Cholesky factorization is explicitly computed
+	 */
+	CGaussianDistributionNew(SGVector<float64_t> mean, float64_t cov,
```

Comment:
These need documentation eventually that explains what parametrisation it corresponds to